### PR TITLE
[Feature][server] Add support for multi sql queries

### LINKED ISSUE #1911: [Feature] Enhancement for Sql task
**Is your feature request related to a problem? Please describe.**
For now, there are 2 limitations of Sql task:
1. can't use variable for other part of sql command, because the limitation of PreparedStatement .
2. can't execute multiple sql command in task content.

**Describe the solution you'd like**
1. Replace variable firstly and then execute sql command with PreparedStatement .
2. Split multiple sql command one by one , and then execute them.

### LINKED ISSUE #3799: [Feature][*] Sql task extension, support multiple sql statement query
*For better global communication, please give priority to using English description, thx! *

*Please review https://dolphinscheduler.apache.org/en-us/docs/development/issue.html when describe an issue.*
**Background**
At present, the SQL task only supports the execution of one statement, that is, the `SQL Statement` in the  picture below currently only support one statement.
image

**Describe the feature**
Support multiple sql statement query
 - Design requirements:
1. Multiple SQL statements input by SQL Statement can be split into multiple statements;
2. Batch execution through jdbc
3. Support multiple data sources oracle, hive, postgresql, mysql

### LINKED ISSUE #4465: [Improvement][SQL component] support multiple lines of sql
**Describe the question**
现在sql组件值只支持单条sql，但希望支持多条（通过例如分号分割）

**What are the current deficiencies and the benefits of improvement**
这里应用场景是说，当sql比较多且联系紧密不可分割的时候，不见得希望每条sql一个流程节点。会过于繁琐。至少在开发阶段希望一起执行和测试。测完了基本功能可以考虑进行适当拆分成多个节点，但也未必到这么细的粒度。

**Which version of DolphinScheduler:**
 -[1.3.4]

**Describe alternatives you've considered**
虽然群里答复说：
"不支持啊, 因为jdbc只能执行一条sql, 如果是多个sql的话,没有办法做到精确拆分.
举例,以分号拆分";"  如果sql里面有注释, 注释里面包含分号";" 这个不是很好处理.
而且有的sql是按照 换行拆分的并不是按照sql拆分的.
这个你可以提个issue , 看看咋处理...
hive的话,推荐使用  shell 节点.  hive -e 执行....
"
但是，
1. 相比直接不支持多行，可以不支持注释，或者不支持注释里有分号，这都可以作为限制，不校验都行，反正写错了运行时就报错呗。
2. shell组件毕竟有一些不便。否则sql组件本身就没有存在的意义了

### LINKED ISSUE #4563: [Feature][SqlTask] Make main sql on sqltask with multi statement
**Describe the feature**
Currently, i saw dolphinscheduler sqltask have presql, mainsql and postsql. But only presql and postsql have multi choice and it not convenient and easy to use. 

**Describe the solution you'd like**
For example, we using pentaho and familar with sql step, it support multi statement with ";" delimeter. It makes we easily copy and paste bunch of sql code to sql step and run it all in once, maybe sqltask can be the same.

**Additional context**
image

### PR DESCRIPTION
**Pull request by Segun-Ogundipe**
## What is the purpose of the pull request

This pull request adds support for multi SQL queries
Fixes #3799

Also close #4563, close #4465, close #1911

## Brief changelog

  - *Add ohdsi repository to pom.xml so maven can find the library*
  - *Add SqlRender dependency to pom.xml*
  - *Add code to SqlTask.java to implement the feature*

### CODE REVIEW

**Comment by Segun-Ogundipe** [comment]
Hi @CalvinKirs, @dailidong, please have a look at this PR. I'm working on fixing the failed test and also increasing the test coverage. Would love to get your feedback on the implementation. 

Huge thanks

File: dolphinscheduler-server/pom.xml [hunk 1]
```diff
@@ -143,6 +162,12 @@
             <groupId>org.springframework</groupId>
             <artifactId>spring-test</artifactId>
         </dependency>
+
+        <dependency>
+            <groupId>org.ohdsi.sql</groupId>
+            <artifactId>SqlRender</artifactId>
+            <version>1.7.0-SNAPSHOT</version>
```
Comment (CalvinKirs) [inline dolphinscheduler-server/pom.xml:169]: Very cool, but I noticed that you have introduced a new license. License is a more important issue. You can follow this document to proceed https://dolphinscheduler.apache.org/en-us/community/development/DS-License .html review.

btw, don't use the snap package

**Comment by davidzollo** [comment]
> Hi @CalvinKirs, @dailidong, please have a look at this PR. I'm working on fixing the failed test and also increasing the test coverage. Would love to get your feedback on the implementation.
> 
> Huge thanks

It's better not to import a new jar here,  just my advice

**Comment by davidzollo** [comment]
hi, long time no reply, I will close this PR temporarily, please submit a new PR if you have any idea

**Pull request closed without merging**